Issue#145 Replace __reactServerTimingStart with browser's navigation start event

TonyHYK opened the pull request:
Now that performance API is more readily available in browsers, we want to use the navigation start event for t0 on the client side. This will allow us to remove the script tag that we write synchronously to log the timing on the client side.

The current proxy for t0 comes down after the head is written. This change will inadvertently increase timings that were relying on `__reactServerTimingStart` with however long it takes to write the first byte (TTFB).

gigabo commented:
@TonyHYK - I like the look of this!

Looks like travis caught a bug: Need to handle undefined `window.performance`.

TonyHYK commented:
Yeah I thought about having a fallback but I wasn't too sure if there's other things out there I can use. Navigation timing is supported by IE9+. So we could potentially not log the timings if the API is not available?
Thoughts?

gigabo commented:
That seems reasonable, though maybe it would be less work to just have a fallback?

Like:

```javascript
window.__reactServerTimingStart = window.performance
    ?window.performance.timing.navigationStart
    :new Date
```

Whatever's easiest.  Not too concerned about IE8. :)

TonyHYK commented:
Ah ok sounds reasonable, I wasn't sure to what extent we want to support old browsers. Fixed and rebased. Also, is there a way to run the build locally?

gigabo commented:
Yep, it's just `npm test` from the repo root.

gigabo commented:
This is great @TonyHYK!  Glad to have that `<script>` out of the beginning of the `<head>`!
